android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -203,14 +219,125 @@ public void handleMessage(Message msg) {
+                        }
+
 
-                        if (googleDai != null) {
-                            pos = googleDai.getContentTime(pos);
-                            duration = googleDai.getContentTime(duration);
+                        if (adsBreakPoints != null) {
+                            for (int i = 0; i < adsBreakPoints.size(); i++) {
+                                final ReadableMap arg_object = adsBreakPoints.getMap(i);
+                                final long adTime = (long) arg_object.getDouble("duration");
+                                long startTime;
+                                long endTime;
+                                final boolean played = arg_object.getBoolean("played");
+                                final boolean started = arg_object.getBoolean("started");
+                                final ReadableArray ads = arg_object.getArray("ads");
+
+
+                                if(adsBreakPoints != null && player.getCurrentManifest() != null && isDrm  && isLive) {
+                                    startTime = (long) (arg_object.getDouble("start") * 1000);
+                                    endTime = startTime + adTime;
+                                } else {
+                                    startTime = (long) Math.floor(arg_object.getDouble("start"));
+                                    endTime = (long) Math.floor(arg_object.getDouble("end")) / 1000;
+                                }
+
+                                if(startTime == _pos  && played && !isLive) {
+                                    player.seekTo(endTime);
+                                    break;
+                                }
+
+
+                                if (startTime == _pos && !started ) {
+                                    WritableMap eventData = Arguments.createMap();
+                                    eventData.putInt("index", i);
+                                    eventEmitter.adEvent("AdBreakStarted", eventData);
+                                }
+
+                                // to end the AD
+                                if (endTime == _pos && !played) {
+                                    Log.d("sleman", "sleman ended");
+
+                                    WritableMap eventData = Arguments.createMap();
+                                    eventData.putInt("index", i);
+                                    eventEmitter.adEvent("AdBreakEnded", eventData);
+
+                                    if(snapBackTimeMs > 0 && !isLive) {
+                                        player.seekTo(getStreamTime(snapBackTimeMs));
+                                        snapBackTimeMs = 0;
+                                    }
+                                    break;
+                                }
+
+                                if(!played) {
+                                    for (int j = 0; j < ads.size(); j++) {
```

Comment:
if ads object is similar to adsBreakPoints , then maybe we should use only one of them
if not and need to keep it as separate object then , follow the same note on  adsBreakPoints.
to convert to java object one in manager